### PR TITLE
PNG: avoid double scaling

### LINKED ISSUE #1464: PNG: Avoid Double-Scaling
Avoid scaling *twice* with non-cubic cells.

By scaling the smaller direction up first we increase the already large png memory. We could instead scale directly with the final scaling factor per direction.

ccing @psychocoderHPC

### PR DESCRIPTION
**Pull request by psychocoderHPC**
close #1464

- avoid that PNGs are scaled two times if `scale_image` is not one and the cell size in PIConGPU is not quadratic
- remove member method `resizeAndScaleImage()`

### CODE REVIEW

File: src/picongpu/include/plugins/output/images/PngCreator.hpp [hunk 1]
```diff
@@ -157,8 +151,8 @@ namespace picongpu
 
         std::stringstream step;
         step << std::setw(6) << std::setfill('0') << header.sim.step;
-        float_X scale_x = header.sim.scale[0];
-        float_X scale_y = header.sim.scale[1];
+        float_X scale_x = header.sim.scale[0] * scale_image;
```
Comment (ax3l) [inline src/picongpu/include/plugins/output/images/PngCreator.hpp]: better write:
```C++
float_X scale_x = scale_image;
float_X scale_y = scale_image;
if( scale_to_cellsize )
{
    scale_x *= header.sim.scale[0];
    scale_y *= header.sim.scale[1];
}
```

File: src/picongpu/include/plugins/output/images/PngCreator.hpp [hunk 2]
```diff
@@ -186,9 +180,6 @@ namespace picongpu
             if ((scale_x != float_X(1.0)) || (scale_y != float_X(1.0)))
```
Comment (ax3l) [inline src/picongpu/include/plugins/output/images/PngCreator.hpp]: careful: the if in the line above should now be removed.

**Comment by ax3l** [comment]
@psychocoderHPC small bug, comments inline.

File: src/picongpu/include/plugins/output/images/PngCreator.hpp (hunk 2 above)
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/output/images/PngCreator.hpp]: You are right I need to handle the case with the scaling to the cell size and without

**Comment by psychocoderHPC** [comment]
updated

**Comment by ax3l** [comment]
now it's wonderful :)

**Pull request merged**